CharlieBackend.Business/Services/AccountService.cs

```diff
@@ -72,7 +72,14 @@ public async Task<Result<AccountDto>> GetAccountCredentialsAsync(AuthenticationD
             {
                 authenticationModel.Password = PasswordHelper.HashPassword(authenticationModel.Password, salt);
 
-                var foundAccount = _mapper.Map<AccountDto>(await _unitOfWork.AccountRepository.GetAccountCredentials(authenticationModel));
+                var accountCredentials = await _unitOfWork.AccountRepository.GetAccountCredentials(authenticationModel);
+
+                if(accountCredentials == null)
+                {
+                    return Result<AccountDto>.GetError(ErrorCode.Unauthorized, "Email or password is incorrect.");
+                }
+
+                var foundAccount = _mapper.Map<AccountDto>(accountCredentials);
 
                 return Result<AccountDto>.GetSuccess(foundAccount);
```

Comment:
Currently, in **3b2e3b6b3d882894f5a6d95eca7268664518197e** (your latest commit by now), this line here is unreachable